feat(curriculum): add string method workshop (#66936)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-string-method/69e20876baed33de636b989a.md

Sometimes, you may want to find the position of a specific character inside a string.

Python provides the built-in `.find()` method for this purpose.

The `.find()` method returns the index (position) of the first occurrence of a substring. For example:

```py
text = 'hello'
e_position = text.find('e')
print('Position: ' + str(e_position))  # Output: Position: 1
```

Notice that `e_position` holds a number (an integer), not a string. To print it with a label, you need to convert it to a string first using `str()`, then join it with the label using `+`.

Now, find the position of the `'@'` symbol in the `email` string.

Create a variable named `at_position` and assign it the result of calling `.find()` on `email`. 

Then, print the `email` variable with the label `'Email: '`.

Also, print the value of `at_position` with the label `'Position of @: '`. Remember to use `str()` to convert `at_position` before concatenating it with the label.

# --hints--

You should create a variable named `at_position`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable('at_position')`)) })
```

You should assign `at_position` using `email.find('@')`.

```js

> ({ test: () => assert(runPython(`'email.find' in _code and 'at_position' in _code`)) })

Reviewer: ```suggestion
({ test: () => assert(runPython(`
node = _Node(_code).find_variable('at_position')
not node.is_empty() and node.is_equivalent("at_position = email.find('@')")
`)) })
```

```

You should print the `email` with the label `'Email: '`.

```js

> ({ test: () => assert(runPython(`'Email: ' in _code and 'email' in _code`)) })

Reviewer: ```suggestion
({ test: () => assert(runPython(`_Node(_code).has_call("print('Email: ' + email)")`)) })
```

```

You should print the `position` with the label `'Position of @: '`.

```js

> ({ test: () => assert(runPython(`'Position of @: ' in _code and 'at_position' in _code`)) })

Reviewer: ```suggestion
({ test: () => assert(runPython(`_Node(_code).has_call("print('Position of @: ' + str(at_position))")`)) })
```

---
id: 69e20876baed33de636b989a
title: Step 3
challengeType: 20
dashedName: step-3
---

# --description--

In the previous step, you assigned a valid email string to the `email` variable.

Sometimes, you may want to find the position of a specific character inside a string.

Python provides the built-in `.find()` method for this purpose.

> The `.find()` method returns the index (position) of the first occurrence of a substring. For example:

Reviewer 2: ```suggestion
Sometimes, you may want to find the position of a specific character inside a string. Python provides the built-in `.find()` method for this purpose. It returns an integer representing the index (position) of the first occurrence of the given substring. For example:
```

```py
text = 'hello'
e_position = text.find('e')

> print('Position: ' + str(e_position))  # Output: Position: 1

Reviewer 2: ```suggestion
text = 'hello'
print(text.find('e'))  # 1
```

```

Notice that `e_position` holds a number (an integer), not a string. To print it with a label, you need to convert it to a string first using `str()`, then join it with the label using `+`.

Now, find the position of the `'@'` symbol in the `email` string.

Create a variable named `at_position` and assign it the result of calling `.find()` on `email`. 

Then, print the `email` variable with the label `'Email: '`.

> Also, print the value of `at_position` with the label `'Position of @: '`. Remember to use `str()` to convert `at_position` before concatenating it with the label.

Reviewer 2: A better use for step 2 would be to print the value of `email`. Therefore the seed for this step would be:
```py
email = 'alice.johnson@company.com'
print('Email: ' + email)
```
```suggestion
Now, call the `.find()` method on `email` and pass it the string `@` to find the position of the at symbol in the `email` string. Assign the result to a variable named `at_position`.

Then, print the result of concatenating the string `Position of @: ` with the value of `at_position`. Remember to use `str()` to convert `at_position` before concatenating it.
```
